Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ContextMenu.cpp

```diff
@@ -65,8 +65,14 @@ namespace EMStudio
                 AZ::Color nodeGroupColor;
                 nodeGroupColor.FromU32(nodeGroup->GetColor());
                 QAction* nodeGroupAction = nodeGroupMenu->addAction(QIcon(new SolidColorIconEngine(nodeGroupColor)), nodeGroup->GetName());
+                nodeGroupAction->setCheckable(true);
                 nodeGroupAction->setData(qulonglong(i + 1)); // index of the menu added, not used
-                connect(nodeGroupAction, &QAction::triggered, this, &BlendGraphWidget::AssignSelectedNodesToGroup);
+                if (currentlyAssignedGroup == nodeGroup) {
+                    nodeGroupAction->setChecked(true);
+                }
+                else {
```

Comment:
The checked action currently doesn't do anything; it's only a visual indicator of what group the node is currently in, which is also apparent from looking at the graph. I think this should either do something useful by removing from the group when clicked, or skip creating the action in this case.

Reply:
Indeed, removing the node from the checked group on click is what I had in mind when adding the checkable action.
Whether that should be implemented at all, or in this patch specifically, is something I would like to discuss first because there's no clear indication of the UX for removing a node from a group.

Reply:
@bhanuja-s Do you have an opinion on this?

Reply:
Implemented here: https://github.com/o3de/o3de/pull/13470